lib/method-map.js

```diff
@@ -0,0 +1,219 @@
+    POST: { command: 'powerAC', payloadParams: { required: ['state'] } }
+  },
+  '/session/:sessionId/appium/device/network_speed': {
+    POST: {
+      command: 'networkSpeed',
+      payloadParams: { required: ['netspeed'] }
+    }
+  },
+  '/session/:sessionId/appium/device/keyevent': {
+    POST: {
+      command: 'keyevent',
+      payloadParams: { required: ['keycode'], optional: ['metastate'] }
+    }
+  },
+  '/session/:sessionId/appium/device/current_activity': { GET: { command: 'getCurrentActivity' } },
+  '/session/:sessionId/appium/device/current_package': { GET: { command: 'getCurrentPackage' } },
+  '/session/:sessionId/appium/device/app_state': {
+    POST: {
+      command: 'queryAppState',
+      payloadParams: { required: [['appId'], ['bundleId']] }
+    }
+  },
+  '/session/:sessionId/appium/device/toggle_airplane_mode': { POST: { command: 'toggleFlightMode' } },
+  '/session/:sessionId/appium/device/toggle_data': { POST: { command: 'toggleData' } },
+  '/session/:sessionId/appium/device/toggle_wifi': { POST: { command: 'toggleWiFi' } },
+  '/session/:sessionId/appium/device/toggle_location_services': { POST: { command: 'toggleLocationServices' } },
+  '/session/:sessionId/appium/device/open_notifications': { POST: { command: 'openNotifications' } },
+  '/session/:sessionId/appium/device/start_activity': {
+    POST: {
+      command: 'startActivity',
+      payloadParams: {
+        required: ['appPackage', 'appActivity'],
+        optional: [
+          'appWaitPackage',
+          'appWaitActivity',
+          'intentAction',
+          'intentCategory',
+          'intentFlags',
+          'optionalIntentArguments',
+          'dontStopAppOnReset'
+        ]
+      }
+    }
+  },
+  '/session/:sessionId/appium/device/system_bars': { GET: { command: 'getSystemBars' } },
+  '/session/:sessionId/appium/device/display_density': { GET: { command: 'getDisplayDensity' } },
+  '/session/:sessionId/appium/app/launch': { POST: { command: 'launchApp' } },
+  '/session/:sessionId/appium/app/close': { POST: { command: 'closeApp' } },
+  '/session/:sessionId/appium/app/reset': { POST: { command: 'reset' } },
+  '/session/:sessionId/appium/app/background': {
+    POST: {
+      command: 'background',
+      payloadParams: { required: ['seconds'] }
+    }
+  },
+  '/session/:sessionId/appium/app/end_test_coverage': {
+    POST: {
+      command: 'endCoverage',
+      payloadParams: { required: ['intent', 'path'] }
+    }
+  },
+  '/session/:sessionId/appium/app/strings': {
+    POST: {
+      command: 'getStrings',
+      payloadParams: { optional: ['language', 'stringFile'] }
+    }
+  },
+  '/session/:sessionId/appium/element/:elementId/value': {
+    POST: {
+      command: 'setValueImmediate',
+      payloadParams: { required: ['text'] }
+    }
+  },
+  '/session/:sessionId/appium/element/:elementId/replace_value': {
+    POST: {
+      command: 'replaceValue',
+      payloadParams: { required: ['text'] }
+    }
+  }
```

Comment:
I assume we have forgotten about push/pull files/folder routes

Reply:
those seemed generically useful enough for any appium driver i decided not to deprecate them.

Reply:
I thought we only keep endpoints that are in W3C spec